// market_exchange_interface.proto
syntax = "proto3";

package market_data;

// Market data service definition
service MarketDataService {
  // Stream market data to subscribers
  rpc SubscribeMarketData (SubscriptionRequest) returns (stream MarketDataUpdate);
}

// Request to subscribe to market data
message SubscriptionRequest {
  string subscriber_id = 1;
  repeated string symbols = 2;
}

// Market data update message
message MarketDataUpdate {
  int64 timestamp = 1;
  repeated SymbolData data = 2;
}

// Data for a single symbol - minute bars
message SymbolData {
  string symbol = 1;
  double open = 2;
  double high = 3;
  double low = 4;
  double close = 5;
  int32 volume = 6;
  int32 trade_count = 7;
  double vwap = 8;
}